bindings/cpp/buffer.cc

```diff
@@ -103,22 +103,70 @@ void StringTable::Clear() {
   strings_to_id_.clear();
 }
 
-EventBuffer::EventBuffer(StringTable* string_table)
+EventBuffer::EventBuffer(StringTable* string_table, size_t chunk_size_bytes)
     : string_table_(string_table) {
-  entries_.clear();
+  if (chunk_size_bytes < kMinimumChunkSizeBytes) {
+    chunk_size_bytes = kMinimumChunkSizeBytes;
+  }
+  chunk_limit_ = chunk_size_bytes / sizeof(uint32_t);
+
+  head_ = current_ = new Chunk(chunk_limit_);
+}
+
+EventBuffer::~EventBuffer() {
+  Chunk* chunk = head_;
+  while (chunk) {
+    Chunk* next_chunk = chunk->next;
+    delete chunk;
+    chunk = next_chunk;
+  }
+}
+
+uint32_t* EventBuffer::ExpandAndAddSlots(size_t count) {
+  Chunk* new_chunk = new Chunk(chunk_limit_);
+  new_chunk->size = count;
+
+  // Publish the final size of the old chunk (atomic release).
```

Comment:
Nope - you're right. I've fixed it. After integrating the new single threaded work that Warren did, I've now got this branch hopelessly unable to merge. I added a torture test and am dropping this PR in favor of directly submitting.